src/js/server/middleware/applications-middleware.ts

```diff
@@ -4,28 +4,71 @@ import * as statuses from 'js/shared/status-constants';
 /**
  * Intercepts the request to check if the user has submitted an application
  * 
- * If they have, it will redirect them to the dashboard. Otherwise, it will let them proceed
+ * If they have, it will redirect them to the previous page. Otherwise, it will let them proceed
  * as normal.
  */
-export function goHomeIfAlreadyApplied(req, res, next) {
-  req.user.getHackerApplication().then((hackerApplication: HackerApplicationInstance) => {
-    if (hackerApplication) {
-      res.redirect(`${req.baseUrl}/dashboard`);
-      return;
-    }
-    next();
+ 
+export function goBackIfApplied(req, res, next) {
+  alreadyApplied(req, res, next).then(applied => {
+    applied === true ? res.redirect('back') : next();
   }).catch(next);
 }
 
 /**
- * Intercepts requests to check if applications are still open, redirecting to the dashboard if not
+ * Intercepts requests to check if applications are still open, redirecting to the previous page
+ * if not
  */
 
-export function checkApplicationsOpen(req, res, next) {
-  if (process.env.APPLICATIONS_OPEN_STATUS === statuses.applicationsOpen.CLOSED) {
-    res.redirect(`${req.baseUrl}/dashboard`);
-    return;
+export function goBackIfApplicationsClosed(req, res, next) {
+  applicationsClosed(req, res, next).then(closed => {
+    closed === true ? res.redirect('back') : next();
+  }).catch(next); 
+}
+
+export function setAppliedStatus(req, res, next) {
+  alreadyApplied(req, res, next).then(applied => {
+    res.locals.applied = applied === true ? true : false;
+    next();
+  }).catch(next); 
+}
+
+export function setApplicationsStatus(req, res, next) {
+  applicationsClosed(req, res, next).then(closed => {
+    res.locals.applicationsOpen = closed === false ? true : false;
+    next();
+  }).catch(next); 
+}
+
+export async function applicationsClosed(req, res, next) {
+  try {
+    if (process.env.APPLICATIONS_OPEN_STATUS === statuses.applicationsOpen.CLOSED) {
+      throw 'Applications closed!';
+    } else {
+      return false;
+    }
+  } catch(error) {
+    if (error === 'Applications closed!') {
+      return true;
+    }
+  }
+}
+
+export async function alreadyApplied(req, res, next) {
+  if (req.user) {
+    return await req.user.getHackerApplication().then((hackerApplication: HackerApplicationInstance) => {
```

Comment:
This seems quite complex.  Would something like `return hackerApplication !== null` work?  Will `getHackerApplication()` throw an exception if the hacker application does not exist and if so are we handling that exception correctly?